js/tyk.js

```diff
@@ -40,7 +40,13 @@ TykJS.TykMiddleware.NewMiddleware.prototype = Object.create(TykJS.TykMiddleware.
 TykJS.TykMiddleware.NewMiddleware.prototype.constructor = TykJS.TykMiddleware.NewMiddleware;
 
 TykJS.TykMiddleware.NewMiddleware.prototype.NewProcessRequest = function(callback) {
-    this.ProcessRequest = callback;
+    if (callback.length == 2) { // backwards compatibility with two-arg type
```

Comment:
Actually, I think it is totally safe to remove this compatibility check. If you pass 3 arguments to the function that have only 2, everything should be fine.